src/network/fragment_array_input_stream.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include <initializer_list>
+#include <vector>
+
+#include "network/fragment_input_stream.h"
+
+namespace lightstep {
+/**
+ * A non-owning stream of fragments.
+ */
+class FragmentArrayInputStream final : public FragmentInputStream {
+ public:
+  FragmentArrayInputStream() noexcept = default;
+
+  explicit FragmentArrayInputStream(std::initializer_list<Fragment> fragments);
+
+  FragmentArrayInputStream& operator=(
+      std::initializer_list<Fragment> fragments);
+
+  // FragmentInputStream
+  int num_fragments() const noexcept override;
+
+  bool ForEachFragment(Callback callback) const noexcept override;
+
+  void Clear() noexcept override;
+
+  void Seek(int fragment_index, int position) noexcept override;
+
+ private:
+  std::vector<Fragment> fragments_;
+
+  int fragment_index_{0};
```

Comment:
It would avoid the cast, but it's also used for arithmetic and it's preferred to use signed types for that: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es102-use-signed-types-for-arithmetic